src/components/Search.tsx

```diff
@@ -0,0 +1,227 @@
+import { Grid, Box, Paper, TextField, FormControl, InputLabel, Select, Button, Typography, Slider, Mark } from '@material-ui/core';
+
+import SearchIcon from '@material-ui/icons/Search';
+
+import { Filters } from '../type-interfaces/Search';
+import { changeFilters, submitSearch } from '../redux/actions/Search';
+import { MIN_COURSE_CODE, MAX_COURSE_CODE, SLIDER_STEP_SIZE } from '../util/UIConstants';
+
+const useStyles = makeStyles((theme: Theme) => createStyles({
+  root: {
+    flexGrow: 1,
+  },
+  paper: {
+    padding: theme.spacing(),
+    textAlign: 'center',
+    color: theme.palette.text.secondary,
+  },
+  queryForm: {
+  },
+  searchContainer: {
+    // margin: '10%',
+  },
+  filterItem: {
+    textAlign: 'center',
+  },
+  letterCodeForm: {
+    margin: theme.spacing(1),
+    width: '100%',
+  },
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+    maxWidth: 300,
+  },
+  chips: {},
+  chip: {},
+  codeTypography: {
+    fontWeight: theme.typography.fontWeightRegular,
+  },
+  searchButton: {
+  },
+}));
+
+
+// REMOVE once we have actual data
+const names = [
+  'ANTH',
+  'ARTI',
+  'CPSC',
+  'OTHE',
+  'STHE',
+  'TTHE',
+];
+
+// Create array of slider marks: MIN_COURSE_CODE to MAX_COURSE_CODE, in steps of SLIDER_STEP_SIZE
+const sliderRange: any[] = Array.from(Array((MAX_COURSE_CODE - MIN_COURSE_CODE) / SLIDER_STEP_SIZE + 1).keys());
+const sliderMarks = sliderRange.map((num, i) => ({
+  value: (MIN_COURSE_CODE + i * SLIDER_STEP_SIZE),
+  label: (MIN_COURSE_CODE + i * SLIDER_STEP_SIZE)+'',
+}));
+
+interface Props {
+  searchFilters: Filters,
+  changeFilters: Function,
+  submitSearch: Function,
+}
+
+const Search = (props: any) => {
+  // eslint-disable-next-line no-shadow
+  const { searchFilters, changeFilters, submitSearch }: Props = props;
+  const classes = useStyles();
+  const searchInputs: Filters = {
+    ...searchFilters,
+  };
+
+  const handleLetterCodeChangeMultiple = (event: React.ChangeEvent<{ value: unknown }>) => {
+    const { options } = event.target as HTMLSelectElement;
+    const codes: string[] = [];
+    for (let i = 0, l = options.length; i < l; i += 1) {
+      if (options[i].selected) {
+        codes.push(options[i].value);
+      }
+    }
+
+    searchInputs.letterCodes = codes;
+    changeFilters(searchInputs);
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    const { value } = event.target as HTMLSelectElement;
+    if (value.trim() === '' || value.trim() === '') {
+      return;
+    }
+    searchInputs.query = value;
+    changeFilters(searchInputs);
+  };
+
+  const handleNumberRangeChange = (_event: any, newValue: number | number[]) => {
+    if (!Array.isArray(newValue) || newValue.length !== 2) {
+      return;
+    }
+    const newRange: number | number[] = newValue;
+    searchInputs.numberRange = newRange;
+    changeFilters(searchInputs);
+  };
+
+  return (
```

Comment:
Would suggest allowing some ipsum lorem text above every bubble, or at least a line of title with a help icon beside it. Longer page may look better than compact page. Otherwise LGTM

Reply:
Nvm I was mistaken, LGTM